lib/cfme/cloud_services/payload_builder.rb

```diff
@@ -0,0 +1,27 @@
+module Cfme
+  module CloudServices
+    module PayloadBuilder
+      INVENTORY_FILE_NAME = "cfme_inventory".freeze
+
+      def gzipped_tar_file_from(payload)
+        require 'rubygems/package'
+        begin
+          packed_file = Tempfile.new(INVENTORY_FILE_NAME)
+          File.open(packed_file.path, "wb") do |file|
+            Zlib::GzipWriter.wrap(file) do |gz|
+              Gem::Package::TarWriter.new(gz) do |tar|
+                json_content = payload
+                tar.add_file_simple(INVENTORY_FILE_NAME, 0o0444, json_content.length) do |io|
+                  io.write(json_content)
+                end
+              end
+            end
+          end
+        ensure
+          yield(packed_file)
```

Comment:
Probably best to yield the file before the ensure block

Reply:
Also thinking it is best to `.close` the file first to ensure it is flushed to disk before passing the path off to another process

Reply:
done 👍